### PR TITLE
fix: Exceptions include detail property for their value 

### PR DESCRIPTION
**Pull request by nicolassanmar**
Exceptions with `detail` but no `message` add that property to the exception value.

Fixes https://github.com/getsentry/sentry-python/issues/2192

### CODE REVIEW

File: sentry_sdk/utils.py [hunk 1]
```diff
@@ -665,6 +665,11 @@ def get_errno(exc_value):
     # type: (BaseException) -> Optional[Any]
     return getattr(exc_value, "errno", None)
 
+def get_error_message(exc_value):
+    # type: (BaseException) -> str
+    return (getattr(exc_value, "message", "") 
+            or getattr(exc_value, "detail", "")
```
Comment (sentrivana) [inline sentry_sdk/utils.py:671]: Let's be safe, just in case:
```suggestion
            or safe_str(getattr(exc_value, "detail", ""))
```

**Review by sentrivana — CHANGES_REQUESTED** [review]
@nicolassanmar Thanks a ton for this PR! The change looks good, left a small comment. Could you please also add a testcase or two for this?

**Comment by nicolassanmar** [comment]
> @nicolassanmar Thanks a ton for this PR! The change looks good, left a small comment. Could you please also add a testcase or two for this?

Sure, been busy this week, but I will update this when I can

**Comment by goharShoukat** [comment]
> > @nicolassanmar Thanks a ton for this PR! The change looks good, left a small comment. Could you please also add a testcase or two for this?
> 
> Sure, been busy this week, but I will update this when I can

@sentrivana I have added test to this PR. 

For some reason, the new commit doesn't show up here and shows on a new branch. 
https://github.com/getsentry/sentry-python/pull/2294

**Comment by sentrivana** [comment]
Thanks @goharShoukat! I'll take a look sometime this week.

**Review by sentrivana — APPROVED** [review]

**Comment by sentrivana** [comment]
@nicolassanmar @goharShoukat Thank you both!

**Comment by goharShoukat** [comment]
> 

Been following the changes you made Ivana. I had a question. The tests I wrote worked in Python 3.10. Is the attempt here to make the code backward compatible too? 

This is my first every open-source contribution, albeit a very small one, I am still quite proud to make it. Looking forward to contributing in the future.

**Comment by sentrivana** [comment]
@goharShoukat It's a great first contribution! Congrats! :)

The change I made to the test had to do with the change I made to the behavior of the `get_error_message` function (this line specifically https://github.com/getsentry/sentry-python/pull/2193/files#diff-e9f88ca6368ce622839e078ed26c741a927752c01ceae64cd504d3ae86de6fd2R689). Before this PR, we were falling back to `safe_str(exc_value)` in case we didn't succeed in getting the `message` attribute, but in the previous version of this PR we would just return an empty string if we didn't find neither `message` nor `detail`, which would have been a breaking change from before. I then had to adjust the test to also check that the fallback is `safe_str(exc_value)` and not an empty string.

Hope that makes sense. Looking forward to more contributions from you in the future!

**Pull request merged**